lib/server/controllers/qcFlagTypes.controller.js

```diff
@@ -106,8 +106,39 @@ const listQcFlagTypesHandler = async (req, res) => {
     }
 };
 
+// eslint-disable-next-line valid-jsdoc
+/**
+ * Update QC Flag Type
+ */
+const updateQcFlagTypeHandler = async (req, res) => {
+    const validatedDTO = await dtoValidator(
+        DtoFactory.bodyOnly({
+            name: Joi.string().optional(),
+            method: Joi.string().optional(),
+            color: Joi.string().regex(/#[0-9a-fA-F]{6}/).optional().allow(null),
+            bad: Joi.boolean().optional(),
+            archived: Joi.boolean().optional().allow(null),
+        }).concat(Joi.object({ params: Joi.object({ id: Joi.number().required() }) })),
```

Comment:
No! The purpose of bodyOnly is to be used exactly when you have only body, if you start to happen params it means you are wrongly using it...

Reply:
refactored